### ISSUE TITLE
Logical focus order between markers and popups

### ISSUE THREAD

**Issue by Malvoz** [Leaflet/Leaflet#8113]
_Raising this issue on behalf of @timveld._

**Is your feature request related to a problem? Please describe.**

The focus order ("tab order") between markers (or other elements bound to popups) and popups is not logical.

This is due to markers and popups existing in different subtrees of the DOM (`leaflet-marker-pane` and `leaflet-popup-pane`).

A [logical focus order is desirable](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html) so that (all) users can:

> encounter information in an order that is consistent with the meaning of the content and can be operated from the keyboard. This reduces confusion by letting users form a consistent mental model of the content.

**Describe the solution you'd like**

@timveld's https://github.com/Leaflet/Leaflet/issues/7968#issuecomment-1069147811:

> - Ensure that the popup appears immediately below the marker in the DOM

**Additional context**

- Current behavior fails WCAG [Success Criterion 2.4.3 Focus Order](https://www.w3.org/TR/WCAG21/#focus-order)
- [web.dev: DOM order matters](https://web.dev/dom-order-matters/)

**Comment by Malvoz** [comment]
Regarding the proposed solution:

> Ensure that the popup appears immediately below the marker in the DOM

A proper DOM order is very important, however I believe it would require a big architectural change that will potentially impact a lot of other things that may seemingly be unrelated at a quick glance, so I'm not sure it is realistic. At least not in the short term.

**Comment by nineplanets** [comment]
I just ran into this same issue while setting up a map and doing accessibility testing.  For me it works fine if there's only one pin on the map, but as soon as you have multiple pins the tab order gets messed up.  Especially if you have more than a couple pins, it's so messed up as to seem completely useless, at first I didn't realize it was just the tab order and thought certain elements were completely inaccessible via tabbing.  The last activity I see here seems to have marked this as "not prioritized."  Is there any sort of plan for when this might be addressed?
